### DISCUSSION TITLE
TomiS: Rationale for CssJs uncurried functions?

### DISCUSSION THREAD

**Discussion by giraud** [SentiaAnalytics/bs-css#228] [General]
Copied from : https://github.com/reasonml-labs/bs-css/issues/225

Hey. I'm trying to convert some of our components to use CssJs module in ReScript syntax. I noticed all of the CssJs functions such as style are defined as uncurried and require the dot to be used when calling. I'm wondering if the requirement for uncurried is absolutely necessary (I don't know, I'm just asking), because the style function is used a lot and every weird quirk (like requiring a dot) makes it that much more cumbersome to use. Also the type hint is ugly with uncurried functions.

**Comment by giraud** [comment]
rationale is here: https://forum.rescript-lang.org/t/bs-css-release-for-better-rescript-js-integration/466

you don't have to use CssJs, you can use Css and fave curried functions

**Comment by TomiS** [comment]
Ah, sorry. I didn't realize I could use discussions. So if I understand correctly, the nice output requires an an uncurried call. Otherwise bucklescript wraps it inside the Curry function that is kinda ugly too, but this time it's on JS side. I start to understand the "uncurried by default" discussion that's been happening

**Comment by giraud** [comment]
I created discussions after I've seen your issue :)
more than uglyness, curried calls have a performance cost that people might want to avoid when applying styles, although it's better to use bs-css for static styles.
